### PR TITLE
Fix: Non-applicable goal in global goals list is no longer clickable

### PR DESCRIPTION
**Pull request by SamuXarick**

### CODE REVIEW

**Comment by SamuXarick** [comment]
Game Script for testing:
[PR 8035 GS test.zip](https://github.com/OpenTTD/OpenTTD/files/4303505/PR.8035.GS.test.zip)
This script creates a global goal that scrolls to the location of a town when clicked.

How to reproduce the bug:
- set up this GS
- start new game
- open global goals window
- click on "-Not Applicable-"
... bug! It scrolls to the location of a town.

File: src/goal_gui.cpp [hunk 1]
```diff
@@ -83,7 +83,7 @@ struct GoalListWindow : public Window {
 		}
 
 		y -= 2; // "Company specific goals:" line.
-		if (y < 0) return;
+		if (y < 0 || this->owner == INVALID_OWNER) return;
 
 		for (const Goal *s : Goal::Iterate()) {
 			if (s->company == this->window_number) {
```
Comment (glx22) [inline src/goal_gui.cpp]: I think it's better to replicate the test used to populate the sublists https://github.com/OpenTTD/OpenTTD/blob/63b56bb76b4090fa3af611cdf74f668a63dc28ed/src/goal_gui.cpp#L209
```suggestion
		if (y < 0) return;

		for (const Goal *s : Goal::Iterate()) {
			if (s->company == this->window_number && s->company != INVALID_COMPANY) {
```

Comment (SamuXarick) [reply src/goal_gui.cpp]: It's not `s->company` that needs to test agains `INVALID_OWNER` / `INVALID_COMPANY`, it's either `this->owner` or `this->window_number`

Comment (SamuXarick) [reply src/goal_gui.cpp]: Nevermind, you're right.

**Review by glx22 — APPROVED** [review]

**Pull request merged**